http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/service/AppServiceInterp.scala

```diff
@@ -1141,7 +1141,7 @@ final class LeoAppServiceInterp[F[_]: Parallel](config: AppServiceConfig,
     samVisibleAppsOpt <- NonEmptyList.fromList(samResources).traverse { apps =>
       authProvider.filterUserVisible(apps, userInfo)
     }
-
+    _ = if (samVisibleAppsOpt.isEmpty) authProvider.isUserEnabled(userInfo)
```

Comment:
`samVisibleAppsOpt` is checked in the below match expression -- maybe put this there.
```
res = samVisibleAppsOpt match {
  case None => Vector.empty  // check enablement
  case Some(apps) if apps.isEmpty => // check enablement
  case Some(apps) => // logic

Reply:
My read was that Some(empty list) wasn't possible because the type is `NonEmptyList`.

Reply:
(and my initial attempt to insert it into the None case was unsuccessful because my Scala instincts are trash - I can move there once I confirm it's working here..)

Reply:
Moved!